sdk/face/azure-ai-vision-face/azure/ai/vision/face/models/_patch.py

```diff
@@ -0,0 +1,56 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+from enum import Enum
+from typing import List
+
+from ._enums import FaceAttributeType
+
+FaceAttributeTypeDetection01 = Enum("FaceAttributeTypeDetection01", [
```

Comment:
@catalinaperalta can you help advise if there is a good way to describe these relationships using typespec?

Reply:
@Han-msft can you give me some examples of the relationships here? If you need an enum to be generated that isnt related to any operation, you would first define it in your typespec, then if it doesnt get generated by default you can use the `@access` generator to mark it as public. Here are the access docs: https://azure.github.io/typespec-azure/docs/libraries/typespec-client-generator-core/reference/decorators#@Azure.ClientGenerator.Core.access

Reply:
If we generate multiple enum, how does strong typed language(C#/Java) accept these different types? Is there inheritance for extensible enum?
Our API looks like
```
op detect {
  image: bytes;
  detectionModel: DetectioModel;
  returnFaceAttributes: FaceAttributeType[]'
}

union DetectionModel {
  string,

  "detection_01",
  "detection_02",
  "detection_03",
}

union FaceAttributeType {
  string,

  "headPose",
  "glaases",
  ...
}|
```
While detection_01 support all attribute, detection_03 only support a few of them, customer will get 400 when sending incompatible attribute.

Reply:
If I understand your scenario correctly, the list of supported values is different between operations, so seems like you should have enums for the specific subsets of supported enum values. Otherwise seems like the typespec might not be fully representing the service.

Reply:
Currently we only have 1 operation for it, `detect(image, recognition_model, detection_model, face_attribute_type)`.
Do you suggest us to have `detect_with_detection_01(image, recognition_model, face_attribute_type_detection_01)`, `detect_with_detection_03(image, recognition_model, face_attribute_type_detection_03)` ...?
It would introduce too many overload for us, since we have 4 recognition model and 3 detection model, we also have 2 kinds of input, image stream and image url.

Reply:
I'm curious what you did for C#? Did you have to add overloads for each operation there? Or how did you support these new groupings? Seems like adding overloads in the typespec would be the best so you also get generated operation signatures in your SDKs that match the values you actually support depending on the model that customers use.

Reply:
In C#, it's define as the same type but under different structure.
```
 public struct Detection01 {
            public static FaceAttributeType HeadPose { get; } = FaceAttributeType.HeadPose;
            public static FaceAttributeType Glasses { get; } = FaceAttributeType.Glasses;
            ...
}
```
https://github.com/Han-msft/azure-sdk-for-net/blob/hachiang/ai-vision-face/sdk/face/Azure.AI.Vision.Face/src/Custom/FaceAttributeType.cs
By doing so, we can keep the same operation for all different models.
```
        public virtual Response<IReadOnlyList<FaceDetectionResult>> DetectFromUrl(
            Uri url,
            FaceDetectionModel detectionModel,
            FaceRecognitionModel recognitionModel,
            bool returnFaceId,
            IEnumerable<FaceAttributeType> returnFaceAttributes = null,
            bool? returnFaceLandmarks = null,
            bool? returnRecognitionModel = null,
            int? faceIdTimeToLive = null,
            CancellationToken cancellationToken = default)
```
https://github.com/Han-msft/azure-sdk-for-net/blob/hachiang/ai-vision-face/sdk/face/Azure.AI.Vision.Face/src/Custom/FaceClient.cs#L101C1-L111C10

Reply:
I see...yeah I dont think typespec currently has support for creating relationships between enums in that way.

Reply:
Of course you could define a model similar to the one you have in C# that would at least give you a clearer service definition. 

You could replace this model: https://github.com/Azure/azure-rest-api-specs/blob/73319147d7508a9621ff1a60fb8123711b04010e/specification/ai/Face/models.detect.tsp#L48

With something like:
```
union Detection03Attributes {
  headPose: "headPose",
  mask: "mask",
  string,
}

model FaceAttributeTypesByModel {
  @doc("The 'detectionModel' associated with the detected faceIds.")
  detection01: FaceAttributeType;

  @doc("The 'recognitionModel' associated with the detected faceIds.")
  detection03: Detection03Attributes;

 // more models here ...
}
```

I would discuss this with the stewardship board so they can weigh in on how to best represent this for your service + give alternate ideas on what the best customer experience would be. (imo the overloads are cleaner and will provide better user experience here)

Reply:
May we keep current implementation and follow up with stewardship board in future release? We'd like to publish current version before Build. It's also the first preview release for our SDK, it should be fine if we add more overload in furutre?

Reply:
Greate, thanks for your feedback @catalinaperalta @kristapratico. I'm going to resolve this conversation. Please let us know if you have any additional questions.